src/rpp/timer.cpp

```diff
@@ -143,6 +252,8 @@ namespace rpp
             win32_sleep_ns(millis * 1'000'000ull);
         #elif __APPLE__ || __linux__ || __EMSCRIPTEN__
             unix_sleep_ns_abstime(millis * 1'000'000ull);
+        #elif RPP_BARE_METAL
+            baremetal::sleep_ms(millis);
```

Comment:
```suggestion
        #elif RPP_FREERTOS
            vTaskDelay(pdMS_TO_TICKS(millis));
        #elif RPP_STM32_HAL
            _stm32_sleep(1'000ull, millis);
```
The current approach has been to keep all platform specific logic inside each function via #if/#elif/#else, this is because maintaining multiple overloads/function prototypes is a large time overhead.

In this case the `baremetal` namespace is not needed, instead it can be implemented at the main implementation site. However, the very platform independent functions like `_sleep()` can just be defined as `static void` functions, so their symbols are limited to this compilation unit.

The final fallback is std::this_thread::sleep_for(), however I'm not sure what the implementation quality is in STM32. We stopped using it ~1 year ago since durations were unreasonably slow in profiler. Now `rpp::TimePoint` and `rpp::Duration` replace it with a single `int64 nsec = 0;` field.

rpp::sleep_ms() and rpp::sleep_us() are also more accurate than std:: alternatives - we had some issues when trying to do <1ms or <=5ms sleeps and these function implementations aim to make it more precise.